Makefile

```diff
@@ -53,11 +61,19 @@ H=$(wildcard *.h */*.h)
 
 all: zsbl.bin fsbl.bin
 
+romgen: $(clk) FPGAzsbl.hex FPGAfsbl.bin
+	cp FPGAzsbl.hex $(BUILD_DIR)/
+	$(rocketchip_dir)/scripts/vlsi_rom_gen $(ROMCONF) $(BUILD_DIR)/FPGAzsbl.hex > $(BUILD_DIR)/rom.v
+
+$(clk): $(dts)
+	awk '/tlclk {/ && !f{f=1; next}; f && match($$0, /^.*clock-frequency.*<(.*)>.*/, arr) { print "#define TL_CLK " arr[1] "UL"}' $< > tl_clock.h
+	cp tl_clock.h $@
+
 elf: zsbl.elf fsbl.elf
 
 asm: zsbl.asm fsbl.asm
 
-lib/version.c: .git/HEAD .git/index
+lib/version.c:
```

Comment:
Were these deps removed accidentally, or was it intentional? Just in case you removed them by accident.

Reply:
Glad to hear your response.

(1) I have tested it on HiFive Unleashed board. It worked normally.

(2) I'm truly happy to do so. Please let me know if you need anything.

And for the removal of **.git/HEAD .git/index** above: yes, they are intentionally

The reason is that when you use the **freedom-u540-c000-bootloader.git** as a submodule inside the **freedom.git**, the HEAD and index files of the **freedom-u540-c000-bootloader.git** will be moved to **freedom/.git/modules/bootrom/freedom-u540-c000-bootloader/**. Thus, this error happened when you try to 'make' from the top directory of the **freedom.git**:

```
~/freedom$ make CONFIG=DevKitU500FPGADesign_WithDevKit125MHz BOOTROM_DIR=`pwd`/bootrom/freedom-u540-c000-bootloader BUILD_DIR=`pwd`/builds/vc707-u500devkit-withpcie-keystoneboot -f Makefile.vc707-u500devkit mcs -j`nproc`
make -C /home/aist-riscv/freedom/bootrom/freedom-u540-c000-bootloader romgen
echo /home/aist-riscv/freedom/rocket-chip/src/main/resources/vsrc/AsyncResetReg.v /home/aist-riscv/freedom/rocket-chip/src/main/resources/vsrc/plusarg_reader.v /home/aist-riscv/freedom/sifive-blocks/vsrc/SRLatch.v /home/aist-riscv/freedom/fpga-shells/xilinx/common/vsrc/PowerOnResetFPGAOnly.v /home/aist-riscv/freedom/fpga-shells/xilinx/vc707/vsrc/sdio.v /home/aist-riscv/freedom/fpga-shells/xilinx/vc707/vsrc/vc707reset.v /home/aist-riscv/freedom/builds/vc707-u500devkit-withpcie-keystoneboot/sifive.freedom.unleashed.DevKitU500FPGADesign_WithDevKit125MHz.rom.v /home/aist-riscv/freedom/builds/vc707-u500devkit-withpcie-keystoneboot/sifive.freedom.unleashed.DevKitU500FPGADesign_WithDevKit125MHz.v > /home/aist-riscv/freedom/builds/vc707-u500devkit-withpcie-keystoneboot/sifive.freedom.unleashed.DevKitU500FPGADesign_WithDevKit125MHz.vsrcs.F
make[1]: Entering directory '/home/aist-riscv/freedom/bootrom/freedom-u540-c000-bootloader'
awk '/tlclk {/ && !f{f=1; next}; f && match($0, /^.*clock-frequency.*<(.*)>.*/, arr) { print "#define TL_CLK " arr[1] "UL"}' /home/aist-riscv/freedom/builds/vc707-u500devkit-withpcie-keystoneboot/sifive.freedom.unleashed.DevKitU500FPGADesign_WithDevKit125MHz.dts > tl_clock.h
dtc /home/aist-riscv/freedom/builds/vc707-u500devkit-withpcie-keystoneboot/sifive.freedom.unleashed.DevKitU500FPGADesign_WithDevKit125MHz.dts -o zsbl/ux00_zsbl.dtb -O dtb
riscv64-unknown-elf-gcc -I. -Ilib/ -O2 -ggdb -march=rv64imafdc -mabi=lp64d -Wall -mcmodel=medany -mexplicit-relocs -DFPGA -o zsbl/main.o -c zsbl/main.c
riscv64-unknown-elf-gcc -I. -Ilib/ -O2 -ggdb -march=rv64imafdc -mabi=lp64d -Wall -mcmodel=medany -mexplicit-relocs -DFPGA -o spi/spi.o -c spi/spi.c
zsbl/ux00_zsbl.dtb: Warning (pci_bridge): Node /soc/pci@2000000000 missing bus-range for PCI bridge
zsbl/ux00_zsbl.dtb: Warning (unit_address_format): Failed prerequisite 'pci_bridge'
zsbl/ux00_zsbl.dtb: Warning (pci_device_reg): Failed prerequisite 'pci_bridge'
zsbl/ux00_zsbl.dtb: Warning (pci_device_bus_num): Failed prerequisite 'pci_bridge'
zsbl/ux00_zsbl.dtb: Warning (simple_bus_reg): Node /soc/tlclk missing or empty reg/ranges property
riscv64-unknown-elf-gcc -I. -Ilib/ -O2 -ggdb -march=rv64imafdc -mabi=lp64d -Wall -mcmodel=medany -mexplicit-relocs -DFPGA -o uart/uart.o -c uart/uart.c
make[1]: *** No rule to make target '.git/HEAD', needed by 'lib/version.c'.  Stop.
make[1]: *** Waiting for unfinished jobs....
cp tl_clock.h /home/aist-riscv/freedom/builds/vc707-u500devkit-withpcie-keystoneboot/sifive.freedom.unleashed.DevKitU500FPGADesign_WithDevKit125MHz.tl_clock.h
make[1]: Leaving directory '/home/aist-riscv/freedom/bootrom/freedom-u540-c000-bootloader'
common.mk:68: recipe for target '/home/aist-riscv/freedom/builds/vc707-u500devkit-withpcie-keystoneboot/sifive.freedom.unleashed.DevKitU500FPGADesign_WithDevKit125MHz.rom.v' failed
make: *** [/home/aist-riscv/freedom/builds/vc707-u500devkit-withpcie-keystoneboot/sifive.freedom.unleashed.DevKitU500FPGADesign_WithDevKit125MHz.rom.v] Error 2
```

Reply:
Thank you for merging. We will make a PR for the documentation of how to build the bistream for vc707 which able to run keystone, preferably this week.

Reply:
Hi @dayeol  and @dkohlbre.
I finally finished the documentation of porting Keystone on local FPGA.
It is pleasure if it could make it in your next release.